ocd/src/target/mod.rs

```diff
@@ -154,5 +146,77 @@ impl From<TargetParseError> for TargetSelectionError {
 
 pub fn identify_target() -> Result<Target, TargetSelectionError> {
     // TODO: Poll this from the connected target. For now return nRF51.
-    Ok(crate::collection::targets::nrf51822::nRF51822())
+    Err(TargetSelectionError::CouldNotAutodetect)
+}
+
+#[derive(Debug, Clone)]
+pub struct FlashAlgorithmContainer {
+    path: String,
+    cache: Option<FlashAlgorithm>,
```

Comment:
```suggestion
    #[serde(default)]
    cache: Option<FlashAlgorithm>,
```

Then you can `#[derive(Deserialize)]`

Reply:
hmm you sure it then can serialize the string into this struct without any more glue code?

Reply:
Right. I guess you'd also need `#[serde(transparent)]`, but I'm not sure that works here.